obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+            # Write to a temporary file then read from it
+            tmp_sfile = NamedTemporaryFile(mode='w', delete=False)
+            for event_line in event_str:
+                tmp_sfile.write(event_line)
+            tmp_sfile.close()
+            with open(tmp_sfile.name) as fh:
+                new_event = _readheader(f=fh)
+                wav_names = _readwavename(f=fh)
+                catalog += _read_picks(f=fh, wav_names=wav_names,
+                                       new_event=new_event)
+            os.remove(tmp_sfile.name)
+            event_str = []
+    f.close()
+    return catalog
+
+
+def _read_picks(f, wav_names, new_event):
+    """Internal pick reader. Use a public function instead.
+
+    :type f: file
+    :param f: File open in read mode
+    :type wav_names: list
+    :param wav_names: List of waveform files in the sfile
+    :type new_event: obspy.core.Event
+    :param new_event: event to associate picks with.
+
+    :returns: populated event.
+    """
+    f.seek(0)
+    evtime = new_event.origins[0].time
+    pickline = []
+    # Set a default, ignored later unless overwritten
+    snr = 999
+    for lineno, line in enumerate(f):
+        if 'headerend' in locals():
+            if len(line.rstrip('\n').rstrip('\r')) in [80, 79] and \
+               (line[79] == ' ' or line[79] == '4' or line[79] == '\n'):
+                pickline += [line]
+        elif line[79] == '7':
+            header = line
+            headerend = lineno
+    amplitude_index = 0
+    for pick_index, line in enumerate(pickline):
+        if line[18:28].strip() == '':  # If line is empty miss it
+            continue
+        station = line[1:6].strip()
+        channel = line[6:8].strip()
+        network = 'NA'  # No network information provided in Sfile.
+        weight = line[14]
+        if weight == '_':
+            phase = line[10:17]
+            weight = 0
+            polarity = ''
+        else:
+            phase = line[10:14].strip()
+            polarity = line[16]
+            if weight == ' ':
+                weight = 0
+        if polarity == '':
+            polarity = "undecidable"
+        elif polarity == 'C':
+            polarity = "positive"
+        elif polarity == 'D':
+            polarity = 'negative'
+        else:
+            polarity = "undecidable"
+        try:
+            time = UTCDateTime(evtime.year, evtime.month, evtime.day,
+                               int(line[18:20]), int(line[20:22]),
+                               int(line[23:28].split('.')[0]),
+                               int(line[23:28].split('.')[1]) * 10000)
+        except (ValueError):
+            time = UTCDateTime(evtime.year, evtime.month, evtime.day,
+                               int(line[18:20]), int(line[20:22]), 0, 0)
+            time += 60  # Add 60 seconds on to the time, this copes with s-file
+            # preference to write seconds in 1-60 rather than 0-59 which
+            # datetime objects accept
```

Comment:
Could you not solve this through explicit logic - might be cleaner and easier to follow, e.g.

```python
y, m, d, h, min, s, msec = ...
offset = 0
if s == 60:
   s = 0
   offset = 60
time = UTCDateTime(...) + offset
```